### PR TITLE
Automate e2e test of the website (basic)

### PR DESCRIPTION
**Pull request by othrayte**
This is an alternative to #376 that strips down the CI step to reduce any potential for unforeseen changes to other CI steps and avoids spamming the ghcr with docker builds.

I decided to test this version to prove if all the extra work the other version does was warranted and it appears if might not have been. The performance of building the container on demand actually seems to be better than building it using the reusable workflow and the only downside is that tests in different browsers would be run sequentially rather than in parallel. In the original PR it turned out however that Cypress seemed to have more to trouble working with firefox than chrome and so I figure it might be better to just test one browser for now for E2E tests as that should be more than sufficient as a smoke test (detecting browser specific failures is not really the goal atm).

Example of tests passing: https://github.com/othrayte/Open-Assistant/pull/6
Example of tests failing: https://github.com/othrayte/Open-Assistant/pull/7

### CODE REVIEW

File: .github/workflows/test-e2e.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        run: docker compose up ci --build -d
+      - name: Run Cypress tests
+        uses: cypress-io/github-action@v5.0.2
+        with:
+          browser: chrome
+          working-directory: website
+      - uses: actions/upload-artifact@v3
+        if: failure() # NOTE: screenshots will be generated only if E2E test failed
+        with:
+          name: cypress-screenshots-${{ matrix.browser }}
```
Comment (AbdBarho) [inline .github/workflows/test-e2e.yaml:31]: `${{ matrix.browser }}` should not be there right?

Comment (othrayte) [reply .github/workflows/test-e2e.yaml]: Fixed.

**Review by yk — CHANGES_REQUESTED** [review]
can we also take `oasst-shared` into the paths?

**Review by yk — APPROVED** [review]
thank you!

**Pull request merged**